src/ng/directive/input.js

```diff
@@ -1180,6 +1186,26 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
   // or form autocomplete on newer browser, we need "change" event to catch it
   element.on('change', listener);
 
+  // Some native input types (date-family) have the ability to change validity without
+  // firing any input/change events.
+  // For these event types, when native validators are present and the browser supports the type,
+  // check for validity changes on various DOM events.
+  if (PARTIAL_VALIDATION_TYPES[type] && ctrl.$$hasNativeValidators && type === attr.type) {
+    element.on(PARTIAL_VALIDATION_EVENTS, function(ev) {
+      if (!timeout) {
```

Comment:
The fact that we are reusing the same timeout with the `else` brach above, is not good I think.
Since the callbacks are conditionally calling `listener`, there might be a timeout registered, so we don't register another one, but that other timeout's callback may end up not calling `listener` (while our callback would).